Training/python/2018v1/Training_v0p1.py

```diff
@@ -218,10 +204,21 @@ def compile_model(model, learning_rate):
     model.compile(loss=TauLosses.tau_crossentropy_v2, optimizer=opt, metrics=metrics, weighted_metrics=metrics)
 
 
-def run_training(train_suffix, model_name, model, data_loader, is_profile):
+def run_training(train_suffix, model_name, model, data_loader, to_profile):
+
+    gpus = tf.config.list_physical_devices('GPU')
+    if gpus:
+        try:
+            tf.config.set_visible_devices(gpus[data_loader.gpu_index], 'GPU')
+            tf.config.experimental.set_virtual_device_configuration(gpus[data_loader.gpu_index],
+                [tf.config.experimental.VirtualDeviceConfiguration(memory_limit=data_loader.gpu_mem*1024)])
+            logical_gpus = tf.config.experimental.list_logical_devices('GPU')
+            print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
+        except RuntimeError as e:
+            print(e)
```

Comment:
Hm... it appears that these settings should be done before `TauLosses.SetSFs(*dataloader.TauLossesSFs)` and `create_model`, otherwise TensorFlow still allocates memory on other GPUs...

Reply:
Unfortunately, I did not test it on machines with several gpus